x-pack/plugins/observability/server/ui_settings.ts

```diff
@@ -358,6 +359,19 @@ export const uiSettings: Record<string, UiSettings> = {
     schema: schema.boolean(),
     requiresPageReload: true,
   },
+  [enableLegacyUptimeApp]: {
+    category: [observabilityFeatureId],
+    name: i18n.translate('xpack.observability.enableLegacyUptimeApp', {
+      defaultMessage: 'Always show legacy Uptime app',
+    }),
+    value: false,
+    description: i18n.translate('xpack.observability.enableLegacyUptimeAppDescription', {
+      defaultMessage:
+        "By default, the legacy Uptime app is hidden from the interface when it doesn't have any data for more than a week. Enable this option to always show it.",
```

Comment:
In the current implementation, Uptime app will be enabled if the index (`heartbeat-*`) exists, so the "more than a week" rule is not correctly implemented.

Also, this rule is not mentioned in the AC of the parent issue.